program/src/main/QuestionSet.vala

```diff
@@ -10,21 +10,31 @@ public class QuestionSet : Object{
     public Gee.ArrayList<Question> questions;
     //private Question currQuestion;
 
-    public QuestionSet(int QNum, double total, double[] b, int pNum, int num){
+    public QuestionSet(int QNum, double total, double[] b, int pNum){
         this.questionNum = QNum;
         this.totalPoints = total;
         this.bounds = b;
         this.pageNum = pNum;
         this.questions = new Gee.ArrayList<Question>();
+        /*
         for(int i = 0; i < num; i++){
             Question q = new Question();
             this.questions.add(q);
         }
+        */
         this.rubricPool = new Gee.HashMap<int, Mark>();
     }
+
+    // Fills question set with new empty questions - HAS to be called before grading starts but can only be called after the number of tests has been determined (aka, end of setup)
```

Comment:
This isn't something we talked about but something we should start doing. Comment functions with valadoc comments. This helps readability and is present in other languages (Javadoc, JSDoc, etc.). To get started you should look [here](https://valadoc.org/markup.htm). If you're using vscode typing '/**' then hitting enter should autogenerate the start of the comment for you

Reply:
Ngl this is pretty cool and should definitely help with the confusion when we eventually pass this off to a PDFGrader 3.0 team and speed up the amount of time it takes to learn how the program works. 

I've added valadoc comments for the new functions `exit_setup()` and `update_bounds()`. It's been a while since I've worked on this feature tbh so I'll update these functions with documentation as I go and get reacquainted with my spaghetti code.